src/components/Calculator.js

```diff
@@ -1,68 +1,43 @@
 import React from 'react';
+import nestedTernary from '../logic/helper';
+import calculate from '../logic/calculate';
 
-// eslint-disable-next-line react/prefer-stateless-function
 class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    const { next, total, operation } = calculate(this.state, event.target.textContent);
+    this.setState({ next, total, operation });
+  }
```

Comment:
- [ ] The click event for all buttons works but the digit is not shown on the display when clicks on it. Please let's fix it.

Reply:
Hi Mih Frederick, thank you for reviewing my code! :smiley: 
I have a few questions if that's okay.
I was trying to introduce this feature earlier before submitting, but I couldn't find a way to do it. That's because it only updates the next property in the state after the click.
I don't know how to explain it well so I'll give an example:
- When I click a digit, I  `console.log(this.state.next)`, and it's null because it only updates when another click happens.
You can see that in demonstration when you click a digit and then an operator. The digit appears on the output.
Can you give me an idea of how to do that feature?